### PR TITLE
User interface is done

### PR DESCRIPTION
**Pull request by lucianosarno**
As the Milestone-2 description asked just for user communication (not for logic yet), we consider that we are done with this step.

We have written the code to check for a draw in milestone 3 (game_logic branch) instead.

Please take a look at it for us? 

Thanks in advance!

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+call_for_play_phrases = []
+call_for_play_phrases << 'So now, PLAYER, could you please choose one of the slots in the board?'
+call_for_play_phrases << 'PLAYER, now it\'s your turn!'
+call_for_play_phrases << 'PLAYER, no more. Choose your slot please.'
+call_for_play_phrases << 'PLAYER, choose your slot and don\'t let your oponent to conquer the world!'
+call_for_play_phrases << 'PLAYER, breath and think twice your next slot, it can be not so easy as you think!'
+call_for_play_phrases << 'PLAYER, what are you waiting for choose your slot??'
+call_for_play_phrases << 'PLAYER, please choose your slot, we don\'t have all the time of the world!!'
+call_for_play_phrases << 'PLAYER, we are waiting for you...'
+call_for_play_phrases << 'PLAYER, please don\'t make us wait more'
```
Comment (bertil291utn) [inline bin/main.rb:66]: ## Split up on different  methods
- Follow these examples

```ruby
def validMove?(player)
 #something to check a valid move
end
```

File: bin/main.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    player = player_one_name.dup
+  else
+    player = player_two_name.dup
+    char = 'O'
+  end
+  puts call_for_play_phrases[rand(call_for_play_phrases.size - 1)].gsub('PLAYER', player)
+  play = catch_play(plays)
+  plays[play - 1] = char
+  puts print_board(plays)
+end
```
Comment (bertil291utn) [inline bin/main.rb]: ## Game does not finish after exactly 9 moves 
- [x] A loop is needed in this game (recommended is `while` but anything else will do) ([OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png))

Take this example

```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it GitHub Flow.

However, in order to accomplish with all requirements, you required to do these changes, listed below:

- [x] Edit pull request message following this [guideline](https://cdn.filestackcontent.com/osvFJEkQHWQsgDe0PJoQ)
- [x] Please next review I'd like to see more @chelmerrox commits, you should participate as navigator and driver

Please add these methods are missing : ([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

* [x] `[REQUIRED]` game tells which player turn it is
* [x] `[REQUIRED]` game asks the player to select from available moves
* [x] `[REQUIRED]` game informs player if the selected move is invalid
* [x] `[REQUIRED]` game displays board after the player move
* [x] `[REQUIRED]` game informs player if the selected move is a winning move
* [x] `[REQUIRED]` game informs player if the selected move is a draw move
* [x] `[REQUIRED]` game repeats all actions for the next player's move
### Summary
- Remember guys your code should not work yet, that's for the next milestone, for this one you could just add as inline comments or mocking up the methods. Example
```ruby
def validMove?(player)
  value=false
  value=true if player.move
value
end
```
-  Don't forget to work with inline comments below this comment box.

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for applying the suggested changes
However, there are some missing requirements

You should add section for these requirements
- `[REQUIRED]` game informs player if the selected move is a winning move
- `[REQUIRED]` game informs player if the selected move is a draw move

Kindly add it and submit a new code review request
Thanks

Mael FOSSO

**Review by euqueme — APPROVED** [review]
> ### Pull Request reviewed by Maria Eugenia Queme P. (TSE)

### Hello @lucianosarno and @chelmerrox :lollipop: :balloon: 

### Thank you for doing the changes requested by the previous reviewers, :medal_sports: :medal_sports: your Tic-tac-toe is working just fine and I always like when players are asked for their names I think this makes it more personal :video_game: now I have a couple of optional suggestions but please after you complete them merge this branch right away (no need to request for a code review again) because you've completed all mandatory requirements Congratulations! :tada: :tada: :tada: 

#### [OPTIONAL]

- [x] The game is letting the players have empty names, let's add a validation so this won't happen
image

- [x] The game is showing this message after a player's turn is over **`check_winner = false`** I believe this is more of a debugging message I personally think you don't need to display this message but if you still want to change it to something more human friendly like: **`We don't have a winner yet!`** 

### That's it, team! :raised_hands: 

### Please :pray: do the changes :technologist: :woman_technologist: and merge this branch right away! :cyclone: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: APPROVED :green_apple:

**Pull request merged**